src/lib.rs

```diff
@@ -45,16 +49,26 @@ pub fn derive(input: TokenStream) -> TokenStream {
         _ => panic!("only named fields are required"),
     };
 
-    let fields_init: Vec<_> = named_fields
-        .named
-        .iter()
-        .map(|f| {
-            let ident = f.ident.as_ref();
-            quote! {
-                #ident: source.#ident,
+    let mut map: HashMap<Ident, FieldModifier> = HashMap::new();
+
+    for f in named_fields.named.iter() {
```

Comment:
use filter instead of for loop to make more readable and collect() into the hash_map directly ?

Reply:
done